### PR TITLE
VRChat: New Precense

### PR DESCRIPTION
**Pull request by KnuffelBeestje**
**Supports:**
- VRChat Portal (vrchat.com)
- VRChat Landing Page (hello.vrchat.com)

**Features:**
- Show username if logged in
- Show search results
- Show the world/avatar/user you're viewing

**Settings:**
- Privacy Mode (Hide your username when logged in. Also doesn't show the avatar/world/user you're currently viewing.)

**hello.vrchat.com**:
image

Discord: 
image

**vrchat.com (portal)**:
image

Discord (Without Privacy Mode turned on): 
image
Discord (With Privacy Mode turned on): 
image

### CODE REVIEW

File: VRChat/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      "hello.vrchat.com"
+    ],
+    "description": {
+      "en": "VRChat is a free-to-play massively multiplayer online virtual reality social platform. It allows players to interact with others as 3D character models.",
+      "nl": "VRChat is een gratis te spelen massaal multiplayer online virtual reality sociaal platform. Hiermee kunnen spelers met anderen communiceren als 3D-personagemodellen."
+    },
+    "service": "VRChat",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/2DIL3Qq.png",
+    "thumbnail": "https://i.imgur.com/0Kf1m7Q.png",
```
Comment (ririxichan) [inline VRChat/dist/metadata.json:18]: There isn't any vrchat banner or could you change it to vrchat site screenshot?

File: VRChat/dist/presence.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                presenceData.state = "Awaiting Authentication";
+                presence.setActivity(presenceData);
+            }
+            else {
+                presenceData.details = "Home Page";
+                presence.setActivity(presenceData);
+            }
+        }
+    }
+});
```
Comment (ririxichan) [inline VRChat/dist/presence.js:189]: delete this file

File: VRChat/presence.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+      }
+    } else if(document.location.hostname == "vrchat.com") {
+        /* Portal */
+        if(privacymode === false) {
+          getUserName();
+          presenceData.state = "User: " + profile;
+        }
+
+
```
Comment (ririxichan) [inline VRChat/presence.ts:126]: delete 1 blank space

File: VRChat/presence.ts [hunk 4]
```diff
@@ -0,0 +1,200 @@
+/* Global variables */
+const presence = new Presence({
+    clientId: "772597423188082729"
+  });
+  var profile: any, title: any, message: any;
+  var browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (ririxichan) [inline VRChat/presence.ts:6]: var -> const, merge consts

File: VRChat/presence.ts [hunk 5]
```diff
@@ -0,0 +1,200 @@
+/* Global variables */
+const presence = new Presence({
+    clientId: "772597423188082729"
+  });
+  var profile: any, title: any, message: any;
+  var browsingStamp = Math.floor(Date.now() / 1000);
+  
+function getUserName(): void { // Get your own username
+  var tempusername = document.querySelector(".user-info > h6");
```
Comment (ririxichan) [inline VRChat/presence.ts:9]: var -> const

File: VRChat/presence.ts [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  
+function getUserName(): void { // Get your own username
+  var tempusername = document.querySelector(".user-info > h6");
+  if(tempusername !== null ) {
+    profile = tempusername.textContent;
+  }
+}
+
+async function getProfileDetails() { // Gets profile from the user you're viewing.
+
```
Comment (ririxichan) [inline VRChat/presence.ts:16]: delete this blank line

File: VRChat/presence.ts [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+async function getProfileDetails() { // Gets profile from the user you're viewing.
+  
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+
+  const privacymode = await presence.getSetting("privacy");
+  const timertoggle = await presence.getSetting("timer");
```
Comment (ririxichan) [inline VRChat/presence.ts:22]: merge consts

File: VRChat/presence.ts [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+  if(timertoggle) {
+    presenceData.startTimestamp = browsingStamp;
+    }
+  var btnfriendcheck = document.querySelector("div.w-100.btn-group-lg.btn-group-vertical > button.btn.btn-primary").textContent;
+  var viewingprofilename = document.querySelector("div.col-md-12 > h2").textContent;
```
Comment (ririxichan) [inline VRChat/presence.ts:28]: var -> const, merge consts

File: VRChat/presence.ts [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+  }
+}
+  /* Main eventHandler */
+presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+      largeImageKey: "logo"
+    };
+
+    const privacymode = await presence.getSetting("privacy");
+    const timertoggle = await presence.getSetting("timer");
```
Comment (ririxichan) [inline VRChat/presence.ts:51]: merge consts

File: VRChat/presence.ts [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+          presenceData.state = "User: " + profile;
+        }
+
+        
+        if(document.location.pathname.includes("/home")) {
+            if(document.location.pathname.includes("/worlds")) {
+                presenceData.details = "Browsing Worlds";
+                presence.setActivity(presenceData);
+            } else if(document.location.pathname.includes("/world")) { /* Viewing a specific world*/
+              var worldname = document.querySelector(".col-md-12 > h3").textContent;
```
Comment (ririxichan) [inline VRChat/presence.ts:132]: var -> const

File: VRChat/presence.ts [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+              presenceData.details = "Viewing Messages";
+              presence.setActivity(presenceData);
+            } else if(document.location.pathname.includes("/user")) { /* Viewing a specific user*/
+              getProfileDetails();
+            }
+            else if(document.location.pathname.includes("/profile")) { /* Viewing Profile*/
+              presenceData.details = "Viewing Profile";
+              presence.setActivity(presenceData);
+            } else if(document.location.pathname.includes("/search")) { /* Searching */
+              var searchresult = window.location.toString().substr(window.location.toString().lastIndexOf('/')+1);
```
Comment (ririxichan) [inline VRChat/presence.ts:154]: var -> const

File: VRChat/presence.ts [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+              presenceData.details = "Searching:";
+              presenceData.state = searchresult;
+              presence.setActivity(presenceData);
+            } else if(document.location.pathname.includes("/avatar")) { /* Viewing a specific avatar*/
+              var avatarname = document.querySelector("div.col-12 > h3").textContent;
+              var avatarpublicstatus = document.querySelector("div.col-12.col-md-8 > h4 > span > small").textContent;
```
Comment (ririxichan) [inline VRChat/presence.ts:160]: var -> const, merge consts

**Review by ririxichan — CHANGES_REQUESTED** [review]
Fix things, also if you would run [Prettier](https://prettier.io) to just make code style better, it would be great

**Review by KnuffelBeestje — COMMENTED** [review]
Everything is working as expected.

**Review by ririxichan — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**